AsyncDisplayKit/ASTextNode.mm

```diff
@@ -237,6 +237,19 @@ - (ASTextKitRenderer *)_renderer
   return [self _rendererWithBounds:self.threadSafeBounds];
 }
 
+- (ASTextKitRenderer *)_rendererWithBoundsSlow:(CGRect)bounds
+{
+  ASDN::MutexLocker l(__instanceLock__);
+
+  CGSize constrainedSize = bounds.size;
+  if (_constrainedSize.width == -INFINITY) {
```

Comment:
Can we call _rendererWithBoundsSlow: from inside _rendererWithBounds:, in the the case the _renderer == nil?

There is one tricky condition to figure out, but it looks like the constrainedSize == INFINITY and _textContainerInset code that is currently duplicated could be unified into one method (just the Slow one).

Also note: the If() here is using _constrainedSize, and I think you _may_ want to use the local variable constrainedSize.